### PR TITLE
[RUNTIME] Shepherd Integration 

### PR DESCRIPTION
**Pull request by brandonxxlee**
Adds tests for the decode API function.

### CODE REVIEW

**Review by baby-bell — CHANGES_REQUESTED** [review]
I don't really like the way we are copy-pasting functions, and it will probably bite us someday, but there's not much we can do for now. I'll wait until Sam takes a look to approve, otherwise LGTM.

**Review by JHGZhou — APPROVED** [review]
I don't fully understand what the arrays in stateManager.py represent (the ones that seem to start with dummy values of 1, 5551, and 1) but I can't really comment on that part very much. Besides that, looks good to me and does what we need it to do

File: runtime/studentAPI.py
Comment (kevinji) [inline runtime/studentAPI.py]: This could just use a normal dict literal; i.e.
```py
func_map = {
    0: identity,
    1: next_power,
    # etc.
}
```

File: runtime/studentAPI.py
Comment (kevinji) [inline runtime/studentAPI.py]: You can shorten this with a lambda:
```py
return lambda input_x: func_a(func_b(input_x))
```

File: runtime/starter_student_code.py
Comment (kevinji) [inline runtime/starter_student_code.py]: These indents should be four spaces instead of tabs.

**Review by kevinji — COMMENTED** [review]
Minor stylistic comments.

File: runtime/studentAPI.py
Comment (baby-bell) [reply runtime/studentAPI.py]: Better yet, an array, since these are integer keys

**Review by baby-bell — APPROVED** [review]

**Pull request merged**
